pkg/config/types/bacalhau.go

```diff
@@ -26,6 +32,15 @@ type Bacalhau struct {
 	DisableAnalytics    bool                `yaml:"DisableAnalytics,omitempty"`
 }
 
+func (b Bacalhau) Validate() error {
+	var err error
+	if !filepath.IsAbs(b.DataDir) {
+		err = errors.Join(err, fmt.Errorf("DataDir (%q) must be an absolute path", b.DataDir))
+	}
+
+	return err
+}
```

Comment:
I understand that relative paths can be confusing and result in unpredictable behaviour, but that shouldn't be the case for `~`. My recommendation is to no attempt to expand it i the config constructor, and validate here that the path is either abs or relative to $HOME

Reply:
At some point we need to expand it before using it for a FS operation, where/when should that expansion happen?